src/main/java/hudson/plugins/sonar/client/WsClient.java

```diff
@@ -116,13 +125,12 @@ public ProjectQualityGate getQualityGateMeasures() throws Exception {
     return null;
   }
 
-  public String getServerVersion(String serverUrl) throws Exception {
+  public String getServerVersion() throws Exception {
     return client.getHttp(serverUrl + API_VERSION, null, null);
   }
 
-  private String getProjectName() throws Exception {
+  public String getProjectName(String projectKey) throws Exception {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00112)